src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -615,6 +615,41 @@ class VM_Version_StubGenerator: public StubCodeGenerator {
   };
 };
 
+uint64_t VM_Version::CPUFeatures_parse(const char *ccstr) {
+  assert(ccstr, "CPUFeatures_parse NULL");
+  if (strcmp(ccstr, "native") == 0) {
+    return Abstract_VM_Version::features();
+  }
+  if (strcmp(ccstr, "generic") == 0) {
+    // 32-bit x86 cannot rely on anything.
+    return 0
+#ifdef AMD64
+      // These options are all in /proc/cpuinfo of one of the first 64-bit CPUs - Atom D2700 (and Opteron 1352): https://superuser.com/q/1572306/1015048
+      | CPU_SSE | CPU_SSE2 // enabled in 'gcc -Q --help=target', used by OpenJDK
+      | CPU_FXSR // enabled in 'gcc -Q --help=target', unused by OpenJDK
+      | CPU_MMX // enabled in 'gcc -Q --help=target', used only by 32-bit x86 OpenJDK
+      | CPU_TSC // not used by gcc, used by OpenJDK
+      | CPU_CX8 // gcc detects it to set cpu "pentium" (=32-bit only), used by OpenJDK
+      | CPU_CMOV // gcc detects it to set cpu "pentiumpro" (=32-bit only), used by OpenJDK
+      | CPU_FLUSH // ="clflush" in cpuinfo, not used by gcc, required by OpenJDK
+#endif
+    ;
+  }
```

Comment:
+1 to this part!  Getting a meaningful default for "generic" and the rationale for why it was chosen into the code is right overall approach.